typeguard.py

```diff
@@ -42,7 +42,8 @@ def __init__(self, func: Callable, frame=None, args: tuple = None,
 
         self.type_hints = _type_hints_map.get(func)
         if self.type_hints is None:
-            hints = get_type_hints(func)
+            frame = inspect.stack()[2][0]
```

Comment:
Why are you ignoring the `frame` object passed to the constructor? This code should **not** rely on the current stack!

Reply:
You're right in most cases. This case though you cannot use the frame that is being extracted from "decoration time" the `class` doesn't exist in that frame yet. My implementation is incomplete. Let me commit the "normal case" first